clickhouse/base/sslsocket.cpp

```diff
@@ -0,0 +1,211 @@
+#include "sslsocket.h"
+
+#include <stdexcept>
+
+#include <openssl/ssl.h>
+#include <openssl/x509v3.h>
+#include <openssl/err.h>
+#include <openssl/asn1.h>
+
+//#include <iostream>
+
+namespace {
+
+std::string getCertificateInfo(X509* cert)
+{
+    std::unique_ptr<BIO, decltype(&BIO_free)> mem_bio(BIO_new(BIO_s_mem()), &BIO_free);
+    X509_print(mem_bio.get(), cert);
+    char * data = nullptr;
+    size_t len = BIO_get_mem_data(mem_bio.get(), &data);
+
+    return std::string(data, len);
```

Comment:
> BIO_get_mem_data() returns the total number of bytes available on success, 0 if b is NULL, or a negative value in case of other errors.

Need to check `len` for error values, I guess.